### PR TITLE
[ENH] : Containerize Frontend

### LINKED ISSUE #26: feat: Containerize the applications
We need to containerize the following applications for an easy deployment:

- [x] frontend
- [ ] backend

### PR DESCRIPTION
**Pull request by Sauradip07**
- Closes #26
**Changes proposed in this pull request:** 
-  Adds two Dockerfiles for development and production environments, and includes the installation of the sharp npm package for image optimization.

- Having separate Dockerfiles for development and production ensures that each environment has the appropriate configurations and dependencies.

- Using sharp for image optimization improves performance and quality of images used within the application.

- Ensuring sharp is installed in standalone mode is crucial for its correct operation in Docker containers.

### CODE REVIEW

File: src/frontend/Dockerfile
Comment (swaptr) [inline src/frontend/Dockerfile:1]: Why do we need the other Dockerfiles besides this one?

File: src/frontend/README.md [hunk 1]
```diff
@@ -11,3 +11,60 @@ yarn dev
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
 
 You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
+
+
+ ##Docker Installation Guide
+
+ This guide provides instructions for setting up and running a Next.js application using Docker. There are two Docker configurations provided: one for development (Dockerfile.dev) and one for production (Dockerfile).
+
+###Prerequisites
```
Comment (swaptr) [inline src/frontend/README.md:20]:
```suggestion
### Prerequisites
```

File: src/frontend/README.md [hunk 2]
```diff
@@ -11,3 +11,60 @@ yarn dev
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
 
 You can start editing the page by modifying `app/page.tsx`. The page auto-updates as you edit the file.
+
+
+ ##Docker Installation Guide
```
Comment (swaptr) [inline src/frontend/README.md:16]:
```suggestion
 ## Docker Installation Guide
```

File: src/frontend/package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
     "react-dom": "^18",
     "react-redux": "^9.1.2",
     "react-textarea-autosize": "^8.5.3",
+    "sharp": "^0.33.4",
```
Comment (swaptr) [inline src/frontend/package.json:26]: Why was this dependency needed? If I recall correctly [Next](https://nextjs.org/docs/pages/building-your-application/optimizing/images) uses sharp by default.

**Review by swaptr — COMMENTED** [review]
This looks good. Some changes are needed and we can get this merged.

File: src/frontend/Dockerfile
Comment (Sauradip07) [reply src/frontend/Dockerfile:1]: I separate the concerns of development and production environments. Each environment has different needs and configurations.
This is the general Dockerfile that can be used for standard builds and might be customized further based on specific requirements. 

`Dockerfile.dev` I have a plan to add features like hot-reloading, which are essential for a smooth development experience.

File: src/frontend/package.json (hunk 3 above)
Comment (Sauradip07) [reply src/frontend/package.json:26]: Yes, you are entirely correct, but for `production build`  it is necessary to have the optional Sharp package. [Link](https://www.reddit.com/r/nextjs/comments/16qx2tg/guys_can_any_one_explain_what_is_sharp_and_why/)

File: src/frontend/Dockerfile
Comment (swaptr) [reply src/frontend/Dockerfile:1]: I understand. For now, we only need one Dockerfile to build the app for production. As I see it, this Dockerfile and Dockerfile.prod will essentially be the same.
For the `Dockerfile.dev`, a better fit for your use case would be to use [Devcontainers](https://code.visualstudio.com/docs/devcontainers/containers), but this is out of the scope of this PR.

**Comment by swaptr** [comment]
LGTM. Thanks for contributing. Merged.

**Pull request merged**